### ISSUE TITLE
linter: Typescript plugin enabled without being listed in plugins if typeAware is set

### ISSUE THREAD

**Issue by Kruptein** [oxc-project/oxc#24291]
### What version of Oxlint are you using?

1.69.0

### What command did you run?

oxlint apps

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
    "categories": {
        "correctness": "error"
    },
    "options": {
        "typeAware": true
    },
    "plugins": ["unicorn", "oxc", "import", "promise"]
}
```

### What happened?

The above config is a minimal reproduction. In our real setup we use an overrides section to add the typescript plugin specifically for all .ts files. This allowed us to have a mixed codebase where typescript rules are applied to .ts files and not to .js files.
The typeAware option can however only be specified on the root, but this was working fine in 1.68.0, since 1.69.0 linting is throwing a bunch of typescript rule errors (e.g. `typescript(await-thenable)`) despite the typescript plugin not being listed.

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue 


I tried to reproduce this with the information provided, but I cannot reproduce TypeScript rules being enabled unless the `typescript` plugin is actually present.

With this config:

```jsonc
{
  "categories": { "correctness": "error" },
  "options": { "typeAware": true },
  "plugins": ["unicorn", "oxc", "import", "promise"]
}
```

and a file containing:

```ts
async function f() { await 1; }
```

`oxlint@1.69.0` reports only `eslint(no-unused-vars)` and `unicorn(no-unnecessary-await)`.

If I add the override shape described in the issue:

```jsonc
{
  "categories": { "correctness": "error" },
  "options": { "typeAware": true },
  "plugins": ["unicorn", "oxc", "import", "promise"],
  "overrides": [
    {
      "files": ["*.ts"],
      "plugins": ["typescript"]
    }
  ]
}
```

then `typescript(await-thenable)` is reported.This is expected: the override enables the TypeScript plugin for `*.ts` files, `categories.correctness` enables correctness rules for enabled plugins, and `options.typeAware: true` allows the type-aware TypeScript rules to run.

Could you share the exact override config from the real project, or a small repo where the minimal config above reports a `typescript(...)` diagnostic without `typescript` appearing in either root `plugins` or a matching override?

Thanks!

**Comment by Kruptein** [comment]
Aha it's due to a nested oxlint extend.

Root oxlint config:
```jsonc
{
    "categories": {
        "correctness": "error"
    },
    "options": {
        "typeAware": true
    },
    "plugins": ["unicorn", "oxc", "import", "promise"]
}
```

make a folder e.g. `test` with in it these two files:

.oxlintrc.json
```jsonc
{
  "extends": ["../.oxlintrc.json"]
}
```

test.js
```js
async function test() {
  const promises = [undefined];
  await Promise.all(promises);
}
```

then run `nx oxlint test/test.js`

```
◄ 0s ○ npx oxlint test/test.js                                                                                                                             node ◫ 24.18.0 10:53

  × eslint(no-unused-vars): Function 'test' is declared but never used.
   ╭─[test/test.js:1:16]
 1 │ async function test() {
   ·                ──┬─
   ·                  ╰── 'test' is declared here
 2 │   const promises = [undefined];
   ╰────
  help: Consider removing this declaration.

  × typescript(await-thenable): Unexpected iterable of non-Promise (non-"Thenable") values passed to promise aggregator.
   ╭─[test/test.js:3:21]
 2 │   const promises = [undefined];
 3 │   await Promise.all(promises);
   ·                     ────┬────
   ·                         ╰── This expression is not Promise-like
 4 │ }
   ╰────
  help: Pass an iterable of Promise-like values, or wrap each synchronous value in `Promise.resolve(...)` before calling the aggregator.

Found 0 warnings and 2 errors.
Finished in 81ms on 1 file using 24 threads.
```

On 1.68.0 only the unused-var is reported.

**Comment by Sysix** [comment]
Your `.oxlintrc.json` config is only:

```
{
  "extends": ["../.oxlintrc.json"]
}
```

which means it enables the default oxlint plugins like:
```
{
  "extends": ["../.oxlintrc.json"],
  "plugins": ["oxc, "typescript", "unicorn"],
}
```

Until a fix, try to use 🤞 :

```
{
  "extends": ["../.oxlintrc.json"],
  "plugins": [],
}
```

**Comment by Kruptein** [comment]
> which means it enables the default oxlint plugins like:

that is indeed the behavior on >=1.69.0, but wasn't in earlier versions. And given the related PR I'm assuming it's not intended either.

> Until a fix, try to use 🤞 :

Ah nice, haven't tried that, though it's not really crucial for me to upgrade above 1.68.0 either, just noticed it during an update session :D
